### ISSUE TITLE
Why copy weight manually instead of using 'load_state_dict'?

### ISSUE THREAD

**Issue by LuChengTHU** [locuslab/deq#7]
Hi, after reading the code detail, I found that you used 'xxx.weight.data = ...' instead of simply using 'xxx.load_state_dict(...)', why?

**Comment by jerrybai1995** [comment]
Hi @LuChengTHU , yes, you are absolutely right and `load_state_dict` is what I use on my current projects to copy the parameters (and using `weight.data` was more of a deprecated effort that I never bothered to update... :-/). However, note that the dropout masks also need to be copied, and that part is not in the state_dict; so we still need to do it explicitly.  

Feel free to send a pull request and update the code there if you wish to do so, I can review and merge the changes.
